src/collector.ts

```diff
@@ -3,230 +3,87 @@
  * Licensed under the Apache-2.0 License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
 'use strict';
```

Comment:
We can rename this file as "common.ts" and move it under new collector folder, this will ensure all collector code in one directory.

Reply:
collector.ts is an interface, collector directory implements it for specific file type. usually generic types should stay at top level., not along side with impl.